### PR TITLE
Add deprecated message to cachegroup/:parameter_id/parameter resp

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue 

## Which Traffic Control components are affected by this PR?

- Traffic Ops

There is not a change to documentation as it is already mark deprecated in there.

## What is the best way to verify this PR?

Start up Perl TO locally with change and make a request to it. On success you should receive back a response that has an alert section with deprecate messaging

## The following criteria are ALL met by this PR

- [x] This PR does not includes tests
- [x] I have explained why documentation is unnecessary
- [x] This PR does include an update to CHANGELOG.md as such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

**Comment by ocket8888** [comment]
I believe deprecations normally have to go through the mailing list. You probably wanna ask about Cache Group Parameters in general.

**Comment by mhoppa** [comment]
@ocket8888 I noticed it was already marked as such in the documentation https://traffic-control-cdn.readthedocs.io/en/latest/api/cachegroup_parameterID_parameter.html

Does that mean it has already gone through mailing list?

**Comment by rawlinp** [comment]
Cache Group Params in general probably can't be deprecated since I'm pretty sure we actually use and depend on them (despite the absurdity of associating parameters to cachegroups).

About the mailing list, we might want to group up all the endpoints we think we should deprecate in 1.1 and discuss the entire set at once on the mailing list. It will probably get pretty messy to discuss the deprecation of API endpoints one at a time on the mailing list. We might get better feedback with a full list too, because users can just take the whole list and check their usage of all the deprecated endpoints at once.

**Comment by ocket8888** [comment]
RE: the docs deprecation stuff: I did that and I shouldn't have. When I was writing that endpoint I'm sure I asked in an internal Slack channel "what is this?" and was told "we don't use that anymore", so I added the deprecation notice without going through the proper procedure.

I like @rawlinp's idea about a bulk deprecation thread. Seems it'll save a lot of time.

File: traffic_ops/app/lib/API/Cachegroup.pm [hunk 1]
```diff
@@ -380,7 +380,7 @@ sub by_parameter_id {
 		}
 	}
 
-	$self->success( { cachegroups => \@data } );
+	$self->success( { cachegroups => \@data }, "API path is deprecated use /cachegroupparameters instead");
```
Comment (ocket8888) [inline traffic_ops/app/lib/API/Cachegroup.pm]: "Success" is a bit of a weird type of message for something that's more of a warning about a deprecation. This should use the "warning" level instead, imo.

Comment (mitchell852) [reply traffic_ops/app/lib/API/Cachegroup.pm]: what are the alert "levels"? success, error, warning and info?

Comment (mitchell852) [reply traffic_ops/app/lib/API/Cachegroup.pm]: i thought you could return an array of alerts. if so, what about 1 success message (for the successful assignment of a param to a cg) and 1 warning message with the deprecation message? or is that weird?

Comment (mhoppa) [reply traffic_ops/app/lib/API/Cachegroup.pm]: No I think that is what we want to do! THat would match the work that was done #3954

**Comment by mhoppa** [comment]
Closing pull request as how we are approaching deprecation is being decided in the mailing list and even if we did decide to deprecate in perl this PR would need some work

**Pull request closed without merging**